packages/server/src/api/controllers/ai/agents.ts

```diff
@@ -57,11 +167,108 @@ export async function remove(ctx: UserCtx<void, void>) {
 export async function fetchHistory(
   ctx: UserCtx<void, FetchAgentHistoryResponse>
 ) {
-  const db = context.getGlobalDB()
+  const db = context.getAppDB()
   const history = await db.allDocs<AgentChat>(
     docIds.getDocParams(DocumentType.AGENT_CHAT, undefined, {
       include_docs: true,
     })
   )
   ctx.body = history.rows.map(row => row.doc!)
 }
+
+export async function fetchToolSources(
+  ctx: UserCtx<void, AgentToolSourceWithTools[]>
+) {
+  const db = context.getAppDB()
+  const toolSources = await db.allDocs<AgentToolSource>(
+    docIds.getDocParams(DocumentType.AGENT_TOOL_SOURCE, undefined, {
+      include_docs: true,
+    })
+  )
+
+  ctx.body = toolSources.rows.map(row => {
+    const doc = row.doc!
+    let tools: any[] = []
+
+    switch (doc.type) {
+      case "BUDIBASE":
+        tools = budibase
+        break
+      case "GITHUB":
+        const ghClient = new GitHubClient(doc.auth?.apiKey)
+        tools = ghClient.getTools()
+        break
+      case "CONFLUENCE":
+        const confluenceClient = new ConfluenceClient(
+          doc.auth?.apiKey,
+          doc.auth?.email,
+          doc.auth?.baseUrl
+        )
+        tools = confluenceClient.getTools()
+        break
+      case "BAMBOOHR":
+        const bamboohrClient = new BambooHRClient(
+          doc.auth?.apiKey,
+          doc.auth?.subdomain,
+        )
+        tools = bamboohrClient.getTools()
+        break
+    }
+
+    return {
+      ...doc,
+      tools,
+    }
+  })
+}
+
+export async function createToolSource(
```

Comment:
We'll probably want to put validation schemas in front of all of these new endpoints. We're trying to get better at not blindly accepting what the frontend gives us and being surprised later to find random stuff in our DB. cc @adrinr who's been championing this idea.